adapters/rack/lib/opentelemetry/adapters/rack/middlewares/tracer_middleware.rb

```diff
@@ -0,0 +1,248 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require_relative '../util/queue_time'
+
+module OpenTelemetry
+  module Adapters
+    module Rack
+      module Middlewares
+        # TODO: implement 'resource name'
+        class TracerMiddleware
+          def initialize(app)
+            @app = app
+          end
+
+          def call(env)
+            # since middleware may only be initialized once (memoized),
+            # but we want to be sure to have a clean slate on each request:
+            dup._call env
```

Comment:
Can we avoid this if you don't rely on global state, in this case, instance variables? I don't see a `dup` in the original DD instrumentation.

Reply:
I've refactored to avoid using instance variables.